src/components/Settings.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+
+import { useState } from 'react';
+import { useTheme } from '../App';
+import * as Api from '../Api';
+import SettingsIcon from "../images/settings-button.png";
+import '../styles/Settings.css';
+
+const Switch = ({ onChange, initialValue, name }) => {
+    const [value, setValue] = useState(initialValue || false);
+
+    return (
+        <div className={ useTheme("switch") }>
+            <input id={ `switch-input-${name.toLowerCase().replace(' ', '_')}` } checked={ value } type="checkbox"
```

Comment:
You can lift this out to make this line shorter.

```js
/* ... */
const snakeCasedName = name.toLowerCase().replace(' ', '_');
/* ... */
```
And change this to use the new variable.
```suggestion
            <input id={ `switch-input-${snakeCasedName}` } checked={ value } type="checkbox"
```